azurerm/resource_arm_dev_test_lab_schedule.go

```diff
@@ -0,0 +1,447 @@
+			return tf.ImportAsExistsError("azurerm_dev_test_schedule", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	tags := d.Get("tags").(map[string]interface{})
+
+	schedule := dtl.Schedule{
+		Location:           &location,
+		ScheduleProperties: &dtl.ScheduleProperties{},
+		Tags:               expandTags(tags),
+	}
+
+	switch status := d.Get("status"); status {
+	case string(dtl.EnableStatusEnabled):
+		schedule.ScheduleProperties.Status = dtl.EnableStatusEnabled
+	case string(dtl.EnableStatusDisabled):
+		schedule.ScheduleProperties.Status = dtl.EnableStatusDisabled
+	default:
+	}
+
+	if taskType := d.Get("task_type").(string); taskType != "" {
+		schedule.ScheduleProperties.TaskType = &taskType
+	}
+
+	if timeZoneId := d.Get("time_zone_id").(string); timeZoneId != "" {
+		schedule.ScheduleProperties.TimeZoneID = &timeZoneId
+	}
+
+	if _, ok := d.GetOk("weekly_recurrence"); ok {
+		weekRecurrence := expandArmDevTestLabScheduleRecurrenceWeekly(d)
+
+		schedule.WeeklyRecurrence = weekRecurrence
+	}
+
+	if _, ok := d.GetOk("daily_recurrence"); ok {
+		dailyRecurrence := expandArmDevTestLabScheduleRecurrenceDaily(d)
+		schedule.DailyRecurrence = dailyRecurrence
+	}
+
+	if _, ok := d.GetOk("hourly_recurrence"); ok {
+		hourlyRecurrence := expandArmDevTestLabScheduleRecurrenceHourly(d)
+
+		schedule.HourlyRecurrence = hourlyRecurrence
+	}
+
+	if _, ok := d.GetOk("notification_settings"); ok {
+		notificationSettings := expandArmDevTestLabScheduleNotificationSettings(d)
+		schedule.NotificationSettings = notificationSettings
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resGroup, devTestLabName, name, schedule)
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(ctx, resGroup, devTestLabName, name, "")
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read  Dev Test Lab Schedule %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmDevTestLabSchedulesRead(d, meta)
+}
+
+func resourceArmDevTestLabSchedulesRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).devTestLabs.LabSchedulesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	devTestLabName := id.Path["labs"]
+	name := id.Path["schedules"]
+
+	resp, err := client.Get(ctx, resGroup, devTestLabName, name, "")
+
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Dev Test Lab Schedule %s: %s", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	d.Set("lab_name", devTestLabName)
+	d.Set("resource_group_name", resGroup)
+
+	if props := resp.ScheduleProperties; props != nil {
+
+		d.Set("time_zone_id", props.TimeZoneID)
+
```

Comment:
Shouldn't we be setting `task_type` here?

Reply:
Thanks, have set it.